### DISCUSSION TITLE
Auth flow not working on hosted Supabase

### DISCUSSION THREAD

**Discussion by DocLabyrinth** [supabase/supabase#5275] [Questions]
I set up a test project on app.supabase.com and tried inviting my other email address as a user. In the case of the initial invite, the password reset email and the magic link email, each link had a redirect_to value which points to localhost:3000 e.g. https://aaa1234.supabase.co/auth/v1/verify?token=token-here&type=recovery&redirect_to=http://localhost:3000

I tried manually changing the redirect to go to the root page of my Supabase panel because this seems to be the app that usually runs on port 3000 if i look in the docker-compose.yml file in the project repo. In this case it follows a chain of redirects which again ends with localhost and lands here: http://localhost:3000/#error_code=404&error_description=User+not+found

I encountered a similar issue when I was testing out the app in docker on my local machine. I worked around this by setting the LAN ip address of my machine as the address for [these lines](https://github.com/supabase/supabase/blob/master/docker/.env.example#L12-L14) in the environment file. However on the local setup I always seem to be logged in as the same user even if I load the app in a different browser. I'm guessing it's using the keys from the environment file as a convenience thing for local development? For now I'll probably switch back to developing on my local machine.

I'm not sure if the problem with the auth flow not working on app.supabase.com is isolated to my app but it's stopping me from using it and evaluating supabase as an option for my project. It seems like Supabase is very promising but this issue is a blocker for me. Do you know if there's a workaround?

**Comment by DocLabyrinth** [comment]
I wondered if I might've accidentally broken something made a fresh project on app.supabase.com and it has the same issue with redirecting to localhost:3000

**Comment by kangmingtay** [comment]
Hey @DocLabyrinth, just checking, are you clicking the link more than once? The invite and recovery links can only be used once, after which the invitation / recovery token gets consumed and clicking on the links subsequently will result in this error being thrown.

**Comment by DocLabyrinth** [comment]
I think I might've clicked some of them twice. That would explain one of the errors but not why it always redirects me to localhost:3000 after the auth flow. It also doesn't seem to solve the issue if I manually paste the app's public hostname in place of localhost:3000 because a chain of redirects happens which still ends up with me redirected to localhost:3000 somehow. It seems like one of the components must be using localhost instead of its public hostname?

**Reply by kangmingtay** [reply]
yeah that's because you'll need to add your custom redirect url to the whitelist which can be found under "Authentication" -> "Settings" -> "Additional redirect URLs" in the Supabase dashboard.

**Comment by DocLabyrinth** [comment]
Sorry it took me a while to get back about this. Setting the base url on the settings page seemed to be what was required. I think I also got a bit confused about how the user/password auth flow worked. I was trying to test it with the Supabase app alone but I think I misunderstood that I needed a UI with a login component set up.

My setup seems to work ok with react-admin/ra-supabase as the UI and with the base url set to the public url for the react-admin app so that the user is redirected there once they've logged in.

**Comment by DocLabyrinth** [comment]
One thing I still can't quite work out: how the UI knows when a user lands on the main page whether they came from an invite link, a password recovery or something similar. I used the <Auth> component from @supabase/ui but when I follow the invite by email flow, I am simply logged in without being given the chance to set a password. This means I won't be able to sign in again next time unless I use a magic link. I looked at several examples but I couldn't see anywhere that had a way to detect it. Do you know if there is some way to set a query string param on the redirect, maybe something like action=invite_accepted or some other way the UI could know to show a dialog so the user can set or reset their password?

**Reply by kangmingtay** [reply]
For password recoveries and invites, you should specify the redirect_url to point to the page/form where you'll provide a password field for the user to set their password. That page would make use of the updateUser function to update the password of the user before continuing with the rest of the app.
